File: python/3492.py
class Solution(object):
    def maxContainers(self, n, w, maxWeight):
        """
        :type n: int
        :type w: int
        :type maxWeight: int
        :rtype: int
        """
        if 1 <= n <= 1000 and 1 <= w <= 1000 and 1 <= maxWeight <= 10**9:
            # total number of available cells
            N = n * n
            Ans = 0
            for i in range(N):
                # I think here use while loop
                if (i+1) * w <= maxWeight:
                    Ans = 1 + i
                else:
                    break
            return Ans
        else:
            return "Entered values crossing limits"

sol = Solution()
n = 2
w = 3 
maxWeight = 15
print(sol.maxContainers(n, w, maxWeight))
n = 3
w = 5 
maxWeight = 25
print(sol.maxContainers(n, w, maxWeight))

File: Javascript/3492.js
// /**
//  * @param {number} n
//  * @param {number} w
//  * @param {number} maxWeight
//  * @return {number}
//  */
// var maxContainers = function(n, w, maxWeight) {
//     if (n >= 1 && n <= 1000 && w >= 1 && w <= 1000 && maxWeight >= 1 && maxWeight <= 10 ** 9) {
//         let N = n * n;
//         for (i=0; i <= N; i++){
//             if(i*w == maxWeight){
//                 return i;
//             }
//         }
//     }
//     else{
//         return "Entered values crossing limits";
//    } 
// };
// n = 2, w = 3, maxWeight = 15;
// console.log("Test Case 1", maxContainers(n, w, maxWeight));
// n = 3, w = 5, maxWeight = 25;
// console.log("Test Case 2", maxContainers(n, w, maxWeight));
// n = 5, w = 20, maxWeight = 200;
// console.log("Test Case 3", maxContainers(n, w, maxWeight));
// n = 4, w = 16, maxWeight = 400;
// console.log("Test Case 4", maxContainers(n, w, maxWeight));



/**
 * @param {number} n
 * @param {number} w
 * @param {number} maxWeight
 * @return {number}
 */
var maxContainers = function(n, w, maxWeight) {
    if (n >= 1 && n <= 1000 && w >= 1 && w <= 1000 && maxWeight >= 1 && maxWeight <= 10 ** 9){
        let N = n * n;
        let possibleContainers = Math.floor(maxWeight / w);
        let Ans = Math.min(possibleContainers, N)
        return Ans
    }  else {
        return "Entered values crossing limits";
    }
};
n = 2, w = 3, maxWeight = 15;
console.log("Test Case 1", maxContainers(n, w, maxWeight));
n = 3, w = 5, maxWeight = 25;
console.log("Test Case 2", maxContainers(n, w, maxWeight));
n = 5, w = 20, maxWeight = 200;
console.log("Test Case 3", maxContainers(n, w, maxWeight));
n = 4, w = 16, maxWeight = 400
console.log("Test Case 4", maxContainers(n, w, maxWeight));
